src/client/handlers/VaultsSecretsRemove.ts

```diff
@@ -30,21 +30,22 @@ class VaultsSecretsDelete extends UnaryHandler<
       );
       const vaultId =
         vaultIdFromName ?? vaultsUtils.decodeVaultId(input.nameOrId);
-      if (vaultId == null) {
-        throw new vaultsErrors.ErrorVaultsVaultUndefined();
-      }
+      if (vaultId == null) throw new vaultsErrors.ErrorVaultsVaultUndefined();
       await vaultManager.withVaults(
         [vaultId],
         async (vault) => {
-          await vaultOps.deleteSecret(vault, input.secretName);
+          for (const secretName of input.secretNames) {
+            await vaultOps.deleteSecret(vault, secretName, {
+              recursive: input.options?.recursive,
+            });
+          }
```

Comment:
It's cool that `deleteSecret` handles all the logic for us so far. But using this way will result in a separate vault log message for each deletion. We want to include all of them within the same vault commit.

To do this you need to modify `vaultsOps.deleteSecret` to take the array of paths and delete each one within the same `vault.writeF()` call. It's a simple change.